src/timeline/Timeline.tsx

```diff
@@ -132,24 +148,30 @@ const Timeline = (props: TimelineProps) => {
 
   const {scrollEvents} = useTimelineOffset({onChangeOffset, scrollOffset, scrollViewRef: scrollView});
 
-  const packedEvents = useMemo(() => {
-    const width = constants.screenWidth - HOURS_SIDEBAR_WIDTH;
-    return populateEvents(events, {screenWidth: width, dayStart: start, overlapEventsSpacing, rightEdgeSpacing});
-  }, [events, start]);
+  const width = constants.screenWidth - HOURS_SIDEBAR_WIDTH;
+
+  const getEvents = (i = 1) => {
+    return populateEvents(pageEvents[i], {
+      screenWidth: width / numberOfDays,
+      dayStart: start,
+      overlapEventsSpacing: overlapEventsSpacing / numberOfDays,
+      rightEdgeSpacing: rightEdgeSpacing / numberOfDays
+    });
+  };
 
   useEffect(() => {
     let initialPosition = 0;
     if (scrollToNow) {
       initialPosition = calcTimeOffset(HOUR_BLOCK_HEIGHT);
-    } else if (scrollToFirst && packedEvents.length > 0) {
-      initialPosition = min(map(packedEvents, 'top')) ?? 0;
+    } else if (scrollToFirst && getEvents().length > 0) {
+      initialPosition = min(map(getEvents(), 'top')) ?? 0;
     } else if (initialTime) {
       initialPosition = calcTimeOffset(HOUR_BLOCK_HEIGHT, initialTime.hour, initialTime.minutes);
     }
 
     if (initialPosition) {
       setTimeout(() => {
-        scrollView?.current?.scrollTo({
+        (Number(numberOfDays) <= 1) && scrollView?.current?.scrollTo({
```

Comment:
What's the purpose of converting `numberOfDays` into a number, it's already a number.. 

Also why not add this condition to the `if (initialPosition) {` check?